docs/08-docker/02-windows-docker-images.mdx

```diff
@@ -17,70 +17,94 @@ found in
-options to avoid this obstacle
+Microsoft does not permit publishing images with VS Build tools which are needed for IL2CPP builds 
+[`discussion`](https://github.com/microsoft/Windows-Containers/issues/102#issuecomment-827170844). 
+You will encounter exceptions in the Unity log similar to the below:
 
-1. Inject/mount Microsoft Visual Studio with VC++ and Windows 10 SDK from the host system into the
-   container during runtime.
-2. Installs vctools, e.g. `choco install visualstudio2017-workload-vctools --no-progress -y` and
-   push it to a private docker registry
+```
+error: Could not set up a toolchain for Architecture x64. Make sure you have the right build tools installed for il2cpp builds. Details:
+IL2CPP C++ code builder is unable to build C++ code. In order to build C++ code for Windows Desktop, you must have one of these installed:
+ * Visual Studio 2022 or newer with C++ compilers and Windows 10 SDK (recommended)
+ * Visual Studio 2019 with C++ compilers and Windows 10 SDK
+ * Visual Studio 2017 with C++ compilers and Windows 10 SDK
+ * Visual Studio 2015 with C++ compilers and Windows 10 SDK
+Visual Studio 2017 (or newer) is detected using `vswhere.exe` as well as VSCOMNTOOLS environment variables.
+Visual Studio 2015 is detected by looking at "SOFTWARE\Microsoft\VisualStudio\14.0_Config\InstallDir" in the registry as well as VSCOMNTOOLS environment variables.
+Windows 10 SDK is detected by looking at "SOFTWARE\Wow6432Node\Microsoft\Microsoft SDKs\Windows\v10.0\InstallationFolder" in the registry.
+Unable to detect any compatible Visual Studio installation!
+ * Found Visual Studio 2022 installation without C++ tool components
+Windows 10 SDK is not installed. You can install from here: https://developer.microsoft.com/en-us/windows/downloads/windows-10-sdk/
+```
 
-### Injecting VC++ into the image
+There are two options to avoid this obstacle:
 
-**To pass VC++ from a host system, two steps are necessary**
+1. (*Recommended*) Inject/mount Microsoft Visual Studio with VC++ and Windows 10 SDK from the host system into the
+   container during runtime.
+2. Build a custom image and push it to a docker registry.
 
-_Add to the system registry (in a container) path and version to Windows SDK_
+#### Option 1. Injecting VC++ from the host system into the container
 
-```powershell
-$windowsSdkDirectory = "HKLM:\SOFTWARE\Wow6432Node\Microsoft\Microsoft SDKs\Windows\v10.0"
+> These instructions have been validated with Visual Studio 2022 and Unity 2021 LTS.
 
-New-Item -Path $windowsSdkDirectory -Force | Out-Null
+_Install Visual Studio Build Tools_
 
-New-ItemProperty -Path $windowsSdkDirectory `
-     -Name InstallationFolder `
-     -PropertyType String `
-     -Force `
-     -Value "C:\Program Files (x86)\Windows Kits\10\" | Out-Null
+The recommended way to install the correct dependencies on the host is with [Chocolatey](https://chocolatey.org/). This works with Windows 10/11, Windows Server, and Windows Server Core.
```

Comment:
Is this really the recommended way in CI?

As a general note; It's best practice for CI has to be deterministic and never randomly break because of downloading a newer package one day.